test/controllers/api/deploy_groups_controller_test.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+require_relative '../../test_helper'
+
+SingleCov.covered!
+
+describe Api::DeployGroupsController do
+  assert_route verb: "GET", path: "/api/deploy_groups", to: "api/deploy_groups#index"
+
+  assert_route verb: "GET", path: "/api/projects/1/stages/2/deploy_groups", \
+               to: "api/deploy_groups#project_deploy_groups", params: {project_id: "1", id: "2"}
+
+  assert_route verb: "PUT", path: "/api/projects/1/stages/2/deploy_groups", \
+               to: "api/deploy_groups#update_deploy_groups", \
+               params: {project_id: "1", id: "2"}
+
+  oauth_setup!
+
+  describe 'when deploy_groups is disabled' do
+    before do
+      DeployGroup.stubs(:enabled?).returns(false)
+      get :index
+    end
+
+    it 'returns precondition_failed' do
+      assert_response :precondition_failed
+    end
+  end
+
```

Comment:
I'd prefer having 1 `it 'returns precondition failed when deploy groups are disabled'` instead of wrapping everything in another layer of describes